sympy/integrals/transforms.py

```diff
@@ -1005,14 +1007,35 @@ def repl(ex, *args):
     expr = repl(expr, Unequality, replue)
     return S(expr)
 
+def expand_dirac_delta(expr):
+    """
+    Expand an expression involving DiractDelta to get it as a linear
+    combination of DiracDelta functions.
+    """
+    return _lin_eq2dict(expr, expr.atoms(DiracDelta))
 
 @_noconds
 def _laplace_transform(f, t, s_, simplify=True):
     """ The backend function for Laplace transforms. """
     from sympy import (re, Max, exp, pi, Min, periodic_argument as arg_,
-                       arg, cos, Wild, symbols, polar_lift)
+                       arg, cos, Wild, symbols, polar_lift, Add)
     s = Dummy('s')
-    F = integrate(exp(-s*t) * f, (t, 0, oo))
+    a = Wild('a', exclude=[t])
+    deltazero = []
+    deltanonzero = []
+    try:
+        integratable, deltadict = expand_dirac_delta(f)
+    except PolyNonlinearError:
+        raise IntegralTransformError(
+        'Laplace', f, 'could not expand DiracDelta expressions')
+    for dirac_func, dirac_coeff in deltadict.items():
+        p = dirac_func.match(DiracDelta(a*t))
+        if p:
+            deltazero.append(dirac_coeff.subs(t,0)/p[a])
+        else:
+            deltanonzero.append(dirac_func*dirac_coeff)
+    F = Add(integrate(exp(-s*t) * Add(integratable, *deltanonzero), (t, 0, oo)),
+            Add(*deltazero))
```

Comment:
What is the output here if I have something like
```python
laplace_transform(DiracDelta(t*(1 - t)), t, s)
```

Reply:
`(1/2 + exp(-s), -oo, True)` :(

So again a DiracDelta(t) is integrated incorrectly.  I think checking whether the argument is zero must be done after all.

Reply:
Maybe we should just match something like `DiracDelta(a*t + b)`. If there is a DiracDelta that doesn't match that then we bail with IntegralTransformError. Otherwise we use something like `p` here:
```python
In [1]: a, b = symbols('a, b')

In [2]: p = Piecewise((0, Ne(b, 0)), (1/a, Ne(a, 0)))

In [3]: p
Out[3]: 
⎧0  for b ≠ 0
⎪            
⎨1           
⎪─  for a ≠ 0
⎩a           

In [4]: p.subs(b, 0)
Out[4]: 
⎧1           
⎨─  for a ≠ 0
⎩a           

In [5]: p.subs(b, 0).subs(a, 0)
Out[5]: nan
```
Should the the condition `Ne(a, 0)` be included in the returned conditions?

Reply:
> In [5]: p.subs(b, 0).subs(a, 0)
> Out[5]: nan
> ```
> 
> Should the the condition `Ne(a, 0)` be included in the returned conditions?

No, because `limit(p.subs(b, 0),a, 0)` is valid and correct.

Reply:
The limit as a goes to zero is not the same as substituting `a=0`. If a and b are both zero then the integrand has a term like `DiracDelta(0)` and the transform can not have any sensible value.

Reply:
OK, you are right. Do you have an example for me how to inluce `Ne(a, 0)` into the returned conditions? I have no prior experience with propagating conditions.

Reply:
I guess where `aux` is returned at the bottom it could be `And(aux, Ne(a, 0))` or something like that. Maybe it's not so important. The result would be `zoo` if you substitute `a=0` anyway...

Reply:
I think it is not so important. If you calculate the LT of `DiracDelta(a*t - b)`, you get `exp(-b*s/a)/a`.

This cannot come into my code unless b isa already 0 and you actually have `DiracDelta(a*t)`, and for that case the limit would be OK. So I think we can lerave this an conclude. OK?

Reply:
Yeah, I guess this is fine.

Reply:
Yippie! .... ehm .... I mean then I can take a breather and then go on to the next topic, inverse LT of rational functions ;)))